saleor/account/events.py

```diff
@@ -71,7 +66,7 @@ def customer_email_changed_event(
 
 
 def customer_placed_order_event(*, user: User, order: Order) -> Optional[CustomerEvent]:
-    if user.is_anonymous:
+    if not user:
```

Comment:
This is the event for customers, so I suppose the assumption was that we don't want to create this event if it wasn't performed by the logged user.

Reply:
Do we have similar events when an app placed an order? I think it would be more consistent to keep these functions here similar, and maybe move this check outside to where this event is called?